server/conn.go

```diff
@@ -1255,6 +1255,9 @@ func (cc *clientConn) dispatch(ctx context.Context, data []byte) error {
 	cc.lastPacket = data
 	cmd := data[0]
 	data = data[1:]
+	vars := cc.ctx.GetSessionVars()
```

Comment:
Do you need to check whether vars is nil and vars.StmtStats is nil?

Reply:
Look like they should never be nil, so no need to check it?